flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/planner/operations/SqlCreateTableConverter.java

```diff
@@ -89,7 +89,10 @@ private CatalogTable createCatalogTable(SqlCreateTable sqlCreateTable) {
         if (sqlCreateTable.getTableLike().isPresent()) {
             SqlTableLike sqlTableLike = sqlCreateTable.getTableLike().get();
             CatalogTable table = lookupLikeSourceTable(sqlTableLike);
-            sourceTableSchema = table.getSchema();
+            sourceTableSchema =
```

Comment:
We should probably remove the `TableSchema` usage here altogether, I just wanted to focus on fixing this first since it's a blocker. Happy to hear whether to do it in this PR or a follow-up refactoring?

Reply:
We can do this in a follow-up issue. This fix is good for now. But you are right that this class should be removed. We should discuss how this will happen. Currently, the `LIKE` is only available to SQL users. If we want to support this for Table API users as well, we should port the utility to `table-api-java`.